File: Okrug360.Places.Api/Entities/Place.cs
using Okrug360.Places.Api.Enums;

namespace Okrug360.Places.Api.Entities
{
    public class Place
    {
        public Guid Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;

        public PlaceCategory Category { get; set; }

        public string Address { get; set; } = string.Empty;

        public string City { get; set; } = "Vranje";

        public double Latitude { get; set; }

        public double Longitude { get; set; }

        public string? ImageUrl { get; set; }

        public DateTime CreatedAt { get; set; }

        public DateTime? UpdatedAt { get;set; }

        public DateTime? PublishedAt {  get; set; }

        public bool IsPublished {  get; set; }
    }
}

File: frontend/src/types/place.ts
export type PlaceCategory =
  | "Restaurant"
  | "Cafe"
  | "Monastery"
  | "Monestery"
  | "Museum"
  | "Park"
  | "CulturalSite"
  | "Government"
  | "Other";

export interface Place {
  id: string;
  name: string;
  description: string;
  category: PlaceCategory;
  address: string;
  city: string;
  latitude: number;
  longitude: number;
  imageUrl?: string | null;
  createdAt: string;
  updatedAt?: string | null;
  publishedAt?: string | null;
  isPublished: boolean;
}

export interface PlaceMapMarker {
  id: string;
  name: string;
  category: PlaceCategory;
  latitude: number;
  longitude: number;
  imageUrl?: string | null;
}

export interface PagedPlacesResponse {
  items: Place[];
  page: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}